ocurrent-plugin/current_ocluster.ml

```diff
@@ -26,6 +26,10 @@ type docker_build = {
   push_target : target option;
 } [@@deriving to_yojson]
 
+type custom = Cluster_api.Custom.t
+
+let custom_to_yojson c = `String (Cluster_api.Custom.kind c)
```

Comment:
I don't think this will work. When the payload changes, we'll need a new database key for it.

So I think we need some kind of `Custom.key` function to extract the important parts. We only need that on the sending side, so it might make sense to split `action` into two types, one for sending and one for receiving. I had a go at that here (just the splitting; not getting the key):

https://github.com/talex5/ocluster/commit/4aa07b2abf11375609ac938205013015d3e85436

Also, this cache key is only needed for the OCurrent plugin, so we might want to make it optional for other users of OCluster.

Reply:
Ah good point, that makes sense. I'll see if I can extend your commit with the key part. Thanks :)) !

Reply:
Thinking about it more, perhaps custom key generation should be part of the `Current_ocluster` plugin's configuration?

Do we need OCurrent support for custom jobs at the moment? If not, that could just be taken out and added later in a separate PR.

Reply:
> How would this function get to the Key module to change how custom jobs are digested? Would we provide a functorised version ...

That sounds reasonable to me. If that turned out to be awkward, another option would be adding the function to the `Key.t` when the key is created (the plugin configuration is available then). A third option is having some kind of registry of handlers for custom jobs.

> in the context of the solver service we'd want to be able to submit jobs in a pipeline for doing the solves

In the case of the existing CIs, there is a pre-existing "analysis" build step that submits jobs to the solver internally, so that wouldn't need to use the plugin directly. In fact, even the existing cluster build jobs don't use the `Current_cluster` builder directly, but instead define their own builder using the low-level `Current_ocluster.Connection` API.